python/modules/TopReweightProducer.py

```diff
@@ -0,0 +1,141 @@
+import ROOT
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+import math
+import numpy as np
+from functools import reduce
+import operator
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection, Object
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoSUSYTools.modules.datamodelRemap import ObjectRemapped, CollectionRemapped
+from PhysicsTools.NanoSUSYTools.modules.SoftBDeepAK8SFProducer import *
+from PhysicsTools.NanoAODTools.postprocessing.tools import deltaPhi, deltaR, closest
+
+class TopReweightProducer(Module):
+    def __init__(self, era, Process, isData = False):
+        self.era = era
+        self.process = Process
+        self.isData = isData
+        self.metBranchName = "MET"
+        self.xsRoot_mg = os.environ["CMSSW_BASE"] + "/src/PhysicsTools/NanoSUSYTools/data/toppt/topPT_MGPowheg_comp.root"
+	self.xsRoot_topptSyst = os.environ["CMSSW_BASE"] + "/src/PhysicsTools/NanoSUSYTools/data/toppt/LostLepton_topPt_systematics.root"
+        self.hist_toppt_up = "topPt_up";
+        self.hist_toppt_dn = "topPt_dn";
+
+    def beginJob(self):
+        pass
+    def endJob(self):
+        pass
+
+    def loadhisto(self,filename,hname):
+        file =ROOT.TFile.Open(filename)
+        hist_ = file.Get(hname)
+        hist_.SetDirectory(0)
+        return hist_
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+        self.topMGPowheg=self.loadhisto(self.xsRoot_mg,self.era)
+        self.topPTSyst_up=self.loadhisto(self.xsRoot_topptSyst,self.hist_toppt_up)
+        self.topPTSyst_dn=self.loadhisto(self.xsRoot_topptSyst,self.hist_toppt_dn)
+        if not self.isData:
```

Comment:
Since we only apply these to ttbar. Maybe you can further constraint them as in ISRWeight : https://github.com/susy2015/NanoSUSY-tools/blob/master/python/modules/ISRWeightProducer.py#L38

Reply:
It's been constrained